nuxeo-compound-documents-web/elements/nuxeo-compound-document-tree.js

```diff
@@ -0,0 +1,480 @@
+            <span><iron-icon icon="icons:chevron-left"></iron-icon></span>
+            <span class="parent">[[item.title]]</span>
+          </a>
+        </template>
+      </div>
+      <nuxeo-tree id="tree" data="[[document]]" controller="[[controller]]" node-key="uid">
+        <template class="horizontal layout">
+          <span role="treeitem" aria-expanded="[[opened]]" class$="[[_computeItemClass(item)]]">
+            <template class="flex" is="dom-if" if="[[!isLeaf]]">
+              <paper-spinner active$="[[loading]]" aria-hidden="true"></paper-spinner>
+              <iron-icon icon="[[_expandIcon(opened)]]" toggle hidden$="[[loading]]" aria-hidden="true"></iron-icon>
+            </template>
+            <template class="flex" is="dom-if" if="[[_isCompoundChild(item)]]">
+              <img src="[[_thumbnail(item)]]" alt$="[[item.title]]" />
+            </template>
+            <span class="node-name flex">
+              <a href$="[[urlFor(item)]]">[[_title(item)]]</a>
+            </span>
+          </span>
+        </template>
+      </nuxeo-tree>
+      <div class="noPermission" hidden$="[[!_noPermission]]">[[i18n('browse.tree.noDocument')]]</div>
+    </div>
+  `,
+
+  is: 'nuxeo-compound-document-tree',
+  behaviors: [RoutingBehavior, I18nBehavior, FiltersBehavior],
+
+  properties: {
+    controller: Object,
+    auto: {
+      type: Boolean,
+      value: false,
+    },
+    rootDocPath: {
+      type: String,
+      value: '/',
+      observer: '_rootDocPathChanged',
+    },
+    docPath: {
+      type: String,
+      value: '/',
+    },
+    document: {
+      type: Object,
+      observer: '_documentChanged',
+    },
+    currentDocument: {
+      type: Object,
+      observer: '_currentDocumentChanged',
+    },
+    parents: {
+      type: Array,
+      value: [],
+    },
+    label: String,
+    visible: {
+      type: Boolean,
+    },
+    cannotSee: {
+      type: Boolean,
+      value: false,
+    },
+    _noPermission: {
+      type: Boolean,
+      value: false,
+    },
+  },
+
+  observers: ['_fetchDocument(docPath, visible)'],
+
+  ready() {
+    window.addEventListener('nuxeo-documents-deleted', (e) => {
+      if (e.detail.documents) {
+        this.removeDocuments(e.detail.documents);
+        return;
+      }
+      // when in select all mode we don't have a list of documents in the event detail
+      this._fetchDocument();
+    });
+    window.addEventListener('document-deleted', (e) => {
+      const { doc } = e.detail;
+      // check if we are deleting a compound child or any document inside the compound
+      if (this._isCompoundChild(doc) || (doc && this.document && this.document.uid !== doc.uid)) {
+        this.removeDocuments([doc]);
+        return;
+      }
+      // if for any reason we don't have a document, we should fetch the current one at least
+      this._fetchDocument();
+    });
+
+    window.addEventListener('refresh-display', () => {
+      this._fetchDocument();
+    });
```

Comment:
For the sake of consistency with the lines below, can't we do it like this?:
```suggestion
    window.addEventListener('refresh-display',  this._fetchDocument.bind(this));
```

Reply:
Sure it makes sense 👍